Northstar.CustomServers/mod/scripts/vscripts/mp/_base_gametype.gnut

```diff
@@ -1412,6 +1412,32 @@ void function CodeCallback_WeaponFireInCloak( entity player )
 		//player.SetCloakFlicker( 1.0, 2.0 )
 
 		DisableCloak( player, 0.5 )
+		entity cloakWeapon
+		int offhandSlot
+		for ( int i = 0; i <= OFFHAND_MELEE; i++ ) // OFFHAND_MELEE is the largest
+		{
+			entity nowWeapon = player.GetOffhandWeapon( i )
+			if( IsValid( nowWeapon ))
+			{
+				if( nowWeapon.GetWeaponClassName() == "mp_ability_cloak" )
+				{
+					cloakWeapon = nowWeapon
+					offhandSlot = i
+				}
+			}
+		}
```

Comment:
Personally I'm not a big fan of this loop in its current state.

In _utility_shared.nut there is a function `entity function GetCloak( entity ent )` which calls `entity function GetOffhand( entity ent, string classname )`

GetOffhand currently only checks the left and right offhand slots, but could be made to check all offhand slots instead, and then you could use GetCloak in your code. Imo this would be a better solution

Reply:
Im doing this loop since I want get which slot player's cloak is in, so that I can give the empty cloak to the right slot

Reply:
respawn resets cloak's `use state` by giving a new one, in vanilla they only checked `OFFHAND_LEFT` so if cloak is in other offhand slots It will remain "using", the yellow bar draining on RUI, causes you to wait a much longer time if you break the cloak early, and it will remove it's mods which sucks for modded cloaks like my `cloak_drone` and `cloak_field`

Reply:
why do you need to give them an empty cloak? `weapon.SetWeaponPrimaryClipCount()` as well as `weapon.SetWeaponPrimaryClipCountAbsolute()` and `weapon.SetWeaponPrimaryAmmoCount()` do none of these work to set the cloak ammo to 0?

Reply:
> why do you need to give them an empty cloak? `weapon.SetWeaponPrimaryClipCount()` as well as `weapon.SetWeaponPrimaryClipCountAbsolute()` and `weapon.SetWeaponPrimaryAmmoCount()` exist, do none of these work to set the cloak ammo to 0?

visual issues, if you try to `weapon.SetWeaponPrimaryClipCountAbsolute()` while it's activating, the progress bar will remain, like this
![cloaking](https://user-images.githubusercontent.com/56738369/194141711-a048c83f-1f4c-4dd0-bdb0-4ed389cb6742.png)

Reply:
and this mean it's in `using` state, which means it cannot be activate until the bar drains out, even if you're using tacical_cdr_on_kill or reset it to full

Reply:
it's also unable to regen ammo itself

Reply:
thats... annoying i have to say

Reply:
there is also `weapon.RegenerateAmmoReset()`? unsure what that does exactly, but i refuse to believe that there is no way for you to fully empty a weapon without having to give a new one

Reply:
> there is also `weapon.RegenerateAmmoReset()`? unsure what that does exactly, but i refuse to believe that there is no way for you to fully empty a weapon without having to give a new one

yea there should be a way, but respawn done decloak by giving a new one, so I did the same and fix their hardcode

Reply:
Ok well I'm going to look into solving that problem at some point maybe, but I'll consider this fine then for now

Reply:
> yea there should be a way, but respawn done decloak by giving a new one, so I did the same and fix their hardcode

Average respawn fix